include/reliable_updates.h

```diff
@@ -0,0 +1,215 @@
+#pragma once
+
+#include <color_spinor_field.h>
+#include <blas_quda.h>
+#include <util_quda.h>
+
+struct reliable_updates {
+
+  const bool alternative_reliable;
+
+  const double u;
+  const double uhigh;
+
+  const double deps;
+  static constexpr double dfac = 1.1;
+  double d_new = 0;
+  double d = 0;
+  double dinit = 0;
+  double xNorm = 0;
+  double xnorm = 0;
+  double pnorm = 0;
+  double ppnorm = 0;
+  double Anorm = 0;
+  double beta = 0.0;
+
+  double rNorm;
+  double r0Norm;
+  double maxrx;
+  double maxrr;
+  double maxr_deflate; // The maximum residual since the last deflation
+  double delta;
+
+  int resIncrease = 0;
+  int resIncreaseTotal = 0;
+  int hqresIncrease = 0;
+  int hqresRestartTotal = 0;
+
+  // this parameter determines how many consective reliable update
+  // residual increases we tolerate before terminating the solver,
+  // i.e., how long do we want to keep trying to converge
+  int maxResIncrease; //  check if we reached the limit of our tolerance
+  int maxResIncreaseTotal;
+
+  bool use_heavy_quark_res;
+
+  int hqmaxresIncrease;
+  int hqmaxresRestartTotal; // this limits the number of heavy quark restarts we can do
+
+  int updateX = 0;
+  int updateR = 0;
+
+  int steps_since_reliable = 1;
+
+  int rUpdate = 0;
+
+  reliable_updates(bool alternative_reliable, double u, double uhigh, double Anorm, double r2, double delta,
```

Comment:
This constructor is a bit painful. I think, to match other design patterns, we should have a `ReliableUpdatesParams` struct that then gets passed to `reliable_updates` (or `ReliableUpdates` as I suggested :) )

Reply:
https://github.com/lattice/quda/pull/1061/commits/aca1ee4c94659aab2be39a304cdd6c32c6408ab9